src/mito2/src/read/scan_region.rs

```diff
@@ -572,6 +576,69 @@ impl ScanInput {
         Ok(sources)
     }
 
+    /// Builds and returns parts to read.
```

Comment:
Does the returned vec have the same length with `partition`?

Reply:
Refactored in https://github.com/GreptimeTeam/greptimedb/pull/3992/commits/78163d791b8f6c327700850f5e107ef6ed49093d

Now it only prune files.